src/Controller/CRUDController.php

```diff
@@ -37,11 +39,21 @@
 use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 use Symfony\Component\Security\Csrf\CsrfToken;
 
+// BC for Symfony < 3.3 where this trait does not exist
+// NEXT_MAJOR: Remove the polyfill and inherit from Controller again
+require_once __DIR__.'/PolyfillControllerTrait.php';
+
 /**
  * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
  */
-class CRUDController extends Controller
+class CRUDController implements ContainerAwareInterface
 {
+
+    // NEXT_MAJOR: Don't use these traits anymore (inherit from Controller instead)
```

Comment:
please use a FQCN instead of "Controller", otherwise it is not clear **which** controller to extend